src/wallet/redux/reducers/__tests__/opening.test.ts

```diff
@@ -0,0 +1,105 @@
+import { walletReducer } from '..';
+
+import * as states from '../../../states';
+import * as actions from '../../actions';
+
+import { itTransitionsToStateType, itDoesntTransition } from './helpers';
+import { scenarios } from '../../../../core';
+
+const {
+  preFundSetupAHex,
+  preFundSetupBHex,
+  preFundSetupASig,
+  preFundSetupBSig,
+  asAddress,
+  bsAddress,
+  asPrivateKey,
+  bsPrivateKey,
+  channelId,
+  participants,
+  channelNonce,
+} = scenarios.standard;
+
+const defaults = { uid: 'uid', displayMode: states.DisplayMode.None, address: asAddress, privateKey: asPrivateKey };
+
+describe('when in WaitForChannel', () => {
+  describe('when we send in a PreFundSetupA', () => { // preFundSetupA is A's move, so in this case we need to be player A
```

Comment:
@tomclose These tests look great! Have we thought about how we are going to test for side effects of state transitions?

Reply:
Currently the only side-effects are (a) sending a message to your opponent and (b) sending a transaction to the blockchain. In reducer-land those side effects are both recorded in the state:

https://github.com/magmo/rps-poc/blob/wallet/src/wallet/states/shared.ts#L9-L10

This means that to check side-effects you just check the state.